ops/Makefile

```diff
@@ -35,9 +49,62 @@ deploy-%-api: .valid-env-% api.tfvars
 	terraform -chdir=$* apply -auto-approve -var-file=../api.tfvars -target="module.simple_report_api"
 
 promote-%: .be-logged-in .valid-env-%
-	az webapp deployment slot swap -g prime-simple-report-$* -n simple-report-api-$* --slot $(SOURCE_SLOT) --target-slot $(TARGET_SLOT)
+	az webapp deployment slot swap -g prime-simple-report-$* -n $(API_NAME) --slot $(SOURCE_SLOT) --target-slot $(TARGET_SLOT)
+
+wait-for-%-slot-commit: .valid-env-%
+	@echo "Starting wait for commit $(DEPLOYED_COMMIT) in $(SOURCE_SLOT) slot of $* at `date`"
+	until  [[ "$(DEPLOYED_COMMIT)"  == "`$(CURL) $(SLOT_INFO_ENDPOINT) | jq -r .git.commit.id`" ]] ; do sleep 5; done;
+	@echo "Finished waiting at `date`"
+
+wait-for-%-commit: .valid-env-%
+	@echo "Starting wait for commit $(DEPLOYED_COMMIT) in $* at `date`"
+	until  [[ "$(DEPLOYED_COMMIT)"  == "`$(CURL) $(INFO_ENDPOINT) | jq -r .git.commit.id`" ]] ; do sleep 5; done;
+	@echo "Finished waiting at `date`"
+
+wait-for-%-release: .valid-env-%
+	@echo "Starting wait for release $(RELEASE_TAG) in $* at `date`"
+	until [[ "$(RELEASE_TAG)"  == "`$(CURL) $(INFO_ENDPOINT) | jq -r .deploy.tag`" ]]; do sleep 5; done
+	@echo "Finished waiting at `date`"
+
+wait-for-%-slot-release: .valid-env-%
+	@echo "Starting wait for release $(RELEASE_TAG) in $* at `date`"
+	until [[ "$(RELEASE_TAG)"  == "`$(CURL) $(SLOT_INFO_ENDPOINT) | jq -r .deploy.tag`" ]]; do sleep 5; done
+	@echo "Finished waiting at `date`"
+
+wait-for-%-slot-readiness: .valid-env-% # could use curl's built-in retry/timeout if we wanted to
+	@echo "Waiting for a successful readiness probe in $*/$(SOURCE_SLOT)"
+	until $(SLOT_READY); do sleep 5; done;
+
+wait-for-%-readiness: .valid-env-% # could use curl's built-in retry/timeout if we wanted to
+	@echo "Waiting for a successful readiness probe in $*"
+	until $(APP_READY); do sleep 5; done;
+
+check-%-slot-commit: .valid-env-%
+	@echo Checking that $(DEPLOYED_COMMIT) is the current deployed commit for $*/$(SOURCE_SLOT)
+	[[ "$(DEPLOYED_COMMIT)"  == "`$(CURL) $(SLOT_INFO_ENDPOINT) | jq -r .git.commit.id`" ]]
+
+check-%-commit: .valid-env-%
+	@echo Checking that $(DEPLOYED_COMMIT) is the current deployed commit for $*
+	[[ "$(DEPLOYED_COMMIT)"  == "`$(CURL) $(INFO_ENDPOINT) | jq -r .git.commit.id`" ]]
+
+check-%-slot-release: .valid-env-%
+	@echo Checking that $(RELEASE_TAG) is the current release on $*/$(SOURCE_SLOT)
+	[[ "$(RELEASE_TAG)"  == "`$(CURL) $(SLOT_INFO_ENDPOINT) | jq -r .deploy.tag`" ]]
+
+check-%-release: .valid-env-%
+	@echo Checking that $(RELEASE_TAG) is the current release on $*
+	[[ "$(RELEASE_TAG)"  == "`$(CURL) $(INFO_ENDPOINT) | jq -r .deploy.tag`" ]]
+
+check-%-slot-readiness: .valid-env-%
+	@echo Checking the readiness probe for $*/$(SOURCE_SLOT)
+	$(SLOT_READY)
+
+check-%-readiness: .valid-env-%
+	@echo Checking the readiness probe for $*
+	$(APP_READY)
 
+# DEPRECATED: This is too greedy a match, replacing it
```

Comment:
What's the plan for this?

Reply:
This is equivalent to `wait-for-%-slot-commit`, it's just too generic a name when we have five other `wait-for-something-somewhere` targets.